jaseci_core/jaseci/prim/node.py

```diff
@@ -52,16 +52,23 @@ def smart_build_fast_edge_ids(self):
         for k in self.fast_edges.keys():
             for v in self.fast_edges[k]:
                 link_order = [v[0], self.jid] if v[1] == FROM else [self.jid, v[0]]
-                edge = Edge(
-                    m_id=self._m_id, h=self._h, kind="edge", name=k, auto_save=False
-                )
-                edge.from_node_id = link_order[0]
-                edge.to_node_id = link_order[1]
-                edge.bidirected = v[1] == BI
-                edge.jid = v[2] if len(v) > 2 else uuid.uuid4().urn
-                edge.context = v[3] if len(v) > 3 else {}
-                edge.save()
-                self._fast_edge_ids.add_obj(edge)
+                edge = self._h.get_obj(self._m_id, v[2])
+                if edge:
+                    v[3] = edge.context
```

Comment:
@amadolid I assume this is to handle cases where a particular fast edge is already built by another node and you are avoiding creating the same Edge object again?

Reply:
Yes sir, tho there's another referencing happen
`v[3] = edge.context`

the last node that got build (`from` or `to`), their fast_edge's context will be used as same context on the created Edge.

> node.py:66 `edge.context = v[3] if len(v) > 3 else {}`
if any change happen on this edge.context it will also be the same value on `v[3]` owner.

> node.py:57 `v[3] = edge.context`
this line will handle the issue. Instead of rebuilding the same Edge we will now try to get it from memory and sync the reference of existing Edge.context to current fast_edge's `v[3]` (context)